app/models/customer.rb

```diff
@@ -0,0 +1,12 @@
+class Customer < ApplicationRecord
+  has_many :rentals
+  has_many :movies, through: :rentals
+
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :postal_code, presence: true
```

Comment:
You're over-validating here. Your app will not break if the phone number or zip code is missing, and spending the time to validate, test the validations, and make sure your other tests respect them is more effort than you should be spending, especially for a school project where working with models is not the core learning goal.